P1684-mdarray/P1684.md

```diff
@@ -334,6 +342,397 @@ In R3 we decided to radically cut back on constructors and simply leave those ou
 
 Finally, `mdarray` does not have the analog of `mdspan`'s constructor that takes an `array<IndexType, N>` of dynamic extents.  This avoids any ambiguity or confusion with a constructor that takes a container instance, in the case where the `container` happens to be an `array<IndexType, N>`.
 
+### Alternative constructor design
+
+We can both significantly reduce the number of constructors *and*
+gain the ability to emplace the container.  This requires
+a bit of shuffling of the arguments, adding a tag and adding a parameter pack.
+
+* Whenever `extents_type` or `mapping_type` is provided, it must come first.
+* Whenever an `mdarray` is constructed with `val`, it must be preceded by a tag.
+* Add a parameter pack `Args...` that essentially emplace constructs the container.
+
+The original adaptors couldn't do this, because variadics didn't exist in C++98.
+We can and should do better now that we have better language facilities.
+
+In the original design,
+the unconstrained `Alloc` parameter is in a weird spot in the constructors
```

Comment:
Why is that "weird"?  Is the problem that this makes it impossible to add a parameter pack to be forwarded for emplace-constructing the container?

Reply:
It is a parameter that goes with the container, but is not right next to the container parameter, like it is for `stack`, `queue` and `priority_queue`.